ClosedXML/Excel/XLWorksheet.cs

```diff
@@ -1173,6 +1174,52 @@ private void ShiftConditionalFormattingColumns(XLRange range, int columnsShifted
             }
         }
 
+        private void ShiftDataValidationColumns(XLRange range, int columnsShifted)
+        {
+            if (!DataValidations.Any()) return;
+            Int32 firstCol = range.RangeAddress.FirstAddress.ColumnNumber;
+            if (firstCol == 1) return;
+
+            int colNum = columnsShifted > 0 ? firstCol - 1 : firstCol;
+            var model = Column(colNum).AsRange();
+
+            foreach (var dv in DataValidations.ToList())
+            {
+                var dvRanges = dv.Ranges.ToList();
+                dv.Ranges.RemoveAll();
+
+                foreach (var dvRange in dvRanges)
+                {
+                    var dvAddress = dvRange.RangeAddress;
+                    IXLRange newRange;
+                    if (dvRange.Intersects(model))
+                    {
+                        newRange = Range(dvAddress.FirstAddress.RowNumber,
+                                         dvAddress.FirstAddress.ColumnNumber,
+                                         dvAddress.LastAddress.RowNumber,
+                                         dvAddress.LastAddress.ColumnNumber + columnsShifted);
```

Comment:
There was an issue with the similar code (it was about a shifting of the conditional formats as I recall) when ranges became invalid after shifting if they included the last row or column. We should trim the result value by `XLHelper.MaxColumnNumber` (`MaxRowNumber` in the `ShiftDataValidationRows` method). An additional test case for this will also be useful.

Reply:
Ah, you're referring to #1042?

Reply:
Exactly